packages/@aws-cdk/aws-lambda-destinations/README.md

```diff
@@ -37,3 +37,89 @@ const myFn = new lambda.Function(this, 'Fn', {
 ```
 
 See also [Configuring Destinations for Asynchronous Invocation](https://docs.aws.amazon.com/lambda/latest/dg/invocation-async.html#invocation-async-destinations).
+
+### Invocation record
+The following example shows the format of the invocation record for a successful invocation:
+```json
+{
+	"version": "1.0",
+	"timestamp": "2019-11-24T23:08:25.651Z",
+	"requestContext": {
+		"requestId": "c2a6f2ae-7dbb-4d22-8782-d0485c9877e2",
+		"functionArn": "arn:aws:lambda:sa-east-1:123456789123:function:event-destinations:$LATEST",
+		"condition": "Success",
+		"approximateInvokeCount": 1
+	},
+	"requestPayload": {
+		"Success": true
+	},
+	"responseContext": {
+		"statusCode": 200,
+		"executedVersion": "$LATEST"
+	},
+	"responsePayload": "<data returned by the function here>"
+}
+```
+
+In case of failure, the record contains the reason and error object:
+```json
+{
+    "version": "1.0",
+    "timestamp": "2019-11-24T21:52:47.333Z",
+    "requestContext": {
+        "requestId": "8ea123e4-1db7-4aca-ad10-d9ca1234c1fd",
+        "functionArn": "arn:aws:lambda:sa-east-1:123456678912:function:event-destinations:$LATEST",
+        "condition": "RetriesExhausted",
+        "approximateInvokeCount": 3
+    },
+    "requestPayload": {
+        "Success": false
+    },
+    "responseContext": {
+        "statusCode": 200,
+        "executedVersion": "$LATEST",
+        "functionError": "Handled"
+    },
+    "responsePayload": {
+        "errorMessage": "Failure from event, Success = false, I am failing!",
+        "errorType": "Error",
+        "stackTrace": [ "exports.handler (/var/task/index.js:18:18)" ]
+    }
+}
+```
+
+#### Destination-specific JSON format
+* For SNS/SQS, the JSON object is passed as the `Message` to the destination.
+* For Lambda, the JSON is passed as the payload to the function.
+* For EventBridge, the JSON is passed as the `Detail` in the PutEvents call. The source is `lambda`,
+and detail type is either `Lambda Function Invocation Result - Success` or `Lambda Function Invocation Result – Failure`.
+The resource fields contain the function and destination ARNs.
+
```

Comment:
Can you add a couple of sentences here on how all this connects to the CDK? As in, when a customer connects `SqsDestination` or `LambdaDestination`, what should they expect from the sqs message or the lambda payload?

Reply:
> * For SNS/SQS, the invocation record JSON is passed as the `Message` to the destination.
> * For Lambda, the invocation record JSON is passed as the payload to the function.
> * For EventBridge, the invocation record JSON ...

Isn't this the same?

Maybe like this?
* For SNS/SQS (`SnsDestionation`/`SqsDestination`), the invocation record JSON is passed as the `Message` to the destination.
* For Lambda (`LambdaDestination`), the invocation record JSON is passed as the payload to the function.
* For EventBridge (`EventBridgeDestination`), the invocation record JSON ...

Reply:
Ah yes, that should be good enough.